mantidproject/mantid
# Indirect Corrections - Sample and Can cross sections for PP and MC

**Description of work.**
This PR adds options to enter the coherent, incoherent and attenuation cross-sections in PaalmanPings and MonteCarlo corrections as an alternative to entering a chemical formula.

**To test:**
**1.**
1.`Interfaces`->`Indirect`->`Corrections`->`Calculate Paalman Pings` tab
2. Load the data below
3. Shape is `Flat Plate`
4. Sample thickness is `0.1cm`
5. In Sample Details make sure Chemical Formula is selected. Enter the formula `H2-O`
6. Click `Run` and then rename the output workspace ending in `_ass`
7. Change Chemical Formula to be `Cross Sections`
8. Enter the coherent, incoherent and attenuation to be `0.039`, `56.052` and `0.222`.
9. Click `Run` and compare the output workspace ending in `_ass` with the workspace previously outputed. The tolerance should be set at 0.001 and there should not be a data mismatch.

**2.**
1.`Interfaces`->`Indirect`->`Corrections`->`Calculate Monte Carlo Absorption` tab
2. Load data below
3. Shape is `Flat Plate`
4. Sample Height and Sample Width is 1cm
5. Sample Thickness is 0.1cm
6. In Sample Details make sure Chemical Formula is selected. Enter the formula `H2-O`
7. Click `Run` and then rename the output workspace ending in `_ass`
8. Change Chemical Formula to be `Cross Sections`
9. Enter the coherent, incoherent and attenuation to be `0.039`, `56.091` and `0.222`.
10. Click `Run` and compare the output workspace ending in `_ass` with the workspace previously outputed.  The tolerance should be set at 0.001 and there should not be a data mismatch.

**Data Files**
[iris26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/3030033/iris26176_graphite002_red.zip)

Fixes #25400

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
